### DISCUSSION TITLE
CameraUI launch() causing app to crash in iOS16

### DISCUSSION THREAD

**Discussion by OldFlexDev** [airsdk/Adobe-Runtime-Support#2444] [General]
I have a pretty old Flex/AIR app that I am trying to maintain. One of the features is the ability to launch the camera and take a photo. Up to iOS15 all was working well but since updating to iOS16 the app immediately crashes when trying to launch the camera. The popup that requests permission to use the camera still appears and i can see in the settings app that camera permission has been granted.

Has anyone else come across this issue?

Thanks

**Comment by ajwfrost** [comment]
My immediate thoughts are to check:
a) do you have the permission string in your info.plist file? `NSCameraUsageDescription` should be added by default, unless you override it (or maybe are doing localised versions?)
b) do you have the camera entitlements added? `com.apple.security.device.camera` needs to be added to the entitlements for signing (via the app descriptor settings)

If you check the log on the device, it may give a better clue: Settings -> Privacy & Security -> Analytics & Improvements -> Analytics Data -> and look for your application name..

thanks

**Reply by OldFlexDev** [reply]
Hi Andew, 

Thank you so much for your response. I do have the useage description set as well as the Photo Library ones.

I'm not so sure about the camera entitlement though - how can I check that? Should it be in the same file as the other iOS settings?

> <iPhone>
        <InfoAdditions><![CDATA[
			<key>UIDeviceFamily</key>
			<array>
				<string>1</string>
				<string>2</string>
			</array>
			<key>UIStatusBarStyle</key>
            <string>UIStatusBarStyleBlackTranslucent</string>
			<key>UISupportedExternalAccessoryProtocols</key>
			<array>
				<string>jp.star-m.starpro</string>
			</array>
			<key>NSAppTransportSecurity</key>
            <dict><key>NSAllowsArbitraryLoads</key><true/></dict>
			<key>NSPhotoLibraryAddUsageDescription</key> 
                <string>Used for adding Photos to your device to include with defects.</string> 
            <key>NSPhotoLibraryUsageDescription</key> 
                <string>Used for selecting Photos from your device to include with defects.</string> 
            <key>NSCameraUsageDescription</key> 
                <string>Used for taking new photos to include with defects.</string> 
		]]></InfoAdditions>
        <requestedDisplayResolution>standard</requestedDisplayResolution>
    </iPhone>

Thank you

Jamie

**Reply by ajwfrost** [reply]
Yep, just alongside those InfoAdditions ones... I think like this:

```
<iPhone>
    <InfoAdditions><![CDATA[
    ... etc ...
      <key>NSCameraUsageDescription</key> 
            <string>Used for taking new photos to include with defects.</string> 
	]]></InfoAdditions>
    <requestedDisplayResolution>standard</requestedDisplayResolution>
    <Entitlements>
        <![CDATA[
            <key>com.apple.security.device.camera</key>
            <true/>
        ]]>
    </Entitlements>
</iPhone>
```
